msu/systems/registry/registry_system.nut

```diff
@@ -1,15 +1,27 @@
 ::MSU.Class.RegistrySystem <- class extends ::MSU.Class.System
 {
+	static ModSources = {};
+	static URLDomain = ::MSU.Class.Enum([
+		"GitHub",
+		"NexusMods"
+	]);
 	Mods = null;
 	constructor()
 	{
 		base.constructor(::MSU.SystemID.Registry);
 		this.Mods = {};
 	}
 
+	function addNewModSource( _modSource )
+	{
+		::MSU.requireInstanceOf(::MSU.Class.ModSource, _modSource.instance());
```

Comment:
Why `_modSource.instance()` and not `_modSource` only?

Reply:
Because you can't `instanceof` against a class, only an instance, `.instance()` generates a dummy instance without calling the constructor

Reply:
No because the `::MSU.requireInstanceOf` already does a `typeof instance` check on the param. So you don't need to do `.instance()`.